src/panels/lovelace/cards/hui-energy-solar-graph-card.ts

```diff
@@ -0,0 +1,412 @@
+import {
+  css,
+  CSSResultGroup,
+  html,
+  LitElement,
+  PropertyValues,
+  TemplateResult,
+} from "lit";
+import { customElement, property, state } from "lit/decorators";
+import { classMap } from "lit/directives/class-map";
+import "../../../components/ha-card";
+import { ChartData, ChartDataset, ChartOptions } from "chart.js";
+import { HomeAssistant } from "../../../types";
+import { LovelaceCard } from "../types";
+import { EnergySolarGraphCardConfig } from "./types";
+import { fetchStatistics, Statistics } from "../../../data/history";
+import {
+  hex2rgb,
+  lab2rgb,
+  rgb2hex,
+  rgb2lab,
+} from "../../../common/color/convert-color";
+import { labDarken } from "../../../common/color/lab";
+import { SolarSourceTypeEnergyPreference } from "../../../data/energy";
+import { isComponentLoaded } from "../../../common/config/is_component_loaded";
+import {
+  ForecastSolarForecast,
+  getForecastSolarForecasts,
+} from "../../../data/forecast_solar";
+import { computeStateName } from "../../../common/entity/compute_state_name";
+import "../../../components/chart/ha-chart-base";
+import "../../../components/ha-switch";
+import "../../../components/ha-formfield";
+
+const SOLAR_COLOR = { border: "#FF9800", background: "#ffcb80" };
+
+@customElement("hui-energy-solar-graph-card")
+export class HuiEnergySolarGraphCard
+  extends LitElement
+  implements LovelaceCard
+{
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @state() private _config?: EnergySolarGraphCardConfig;
+
+  @state() private _data?: Statistics;
+
+  @state() private _chartData?: ChartData;
+
+  @state() private _forecasts?: Record<string, ForecastSolarForecast>;
+
+  @state() private _chartOptions?: ChartOptions;
+
+  @state() private _showAllForecastData = false;
+
+  private _fetching = false;
+
+  private _interval?: number;
+
+  public disconnectedCallback() {
+    super.disconnectedCallback();
+    if (this._interval) {
+      clearInterval(this._interval);
+      this._interval = undefined;
+    }
+  }
+
+  public connectedCallback() {
+    super.connectedCallback();
+    if (!this.hasUpdated) {
+      return;
+    }
+    this._getStatistics();
+    // statistics are created every hour
+    clearInterval(this._interval);
+    this._interval = window.setInterval(
+      () => this._getStatistics(),
+      1000 * 60 * 60
```

Comment:
Statistics are created every hour at 12 past whole. In the future let's schedule an interval at 20 past or something.